dts: bindings: dac: Change the property names in the overlay

rruuaanng opened the pull request:
Change the property names in the bindings and overlay
to use hyphens(-) for separation instead of underscores(_).

rruuaanng committed:
dts: bindings: dac: Change the property names in the overlay

Change the property names in the bindings and overlay
to use hyphens(-) for separation instead of underscores(_).

Signed-off-by: James Roy <rruuaanng@outlook.com>

martinjaeger commented on doc/releases/migration-guide-4.1.rst:
```diff
@@ -157,6 +157,12 @@ Display
+DAC
+===
+
+* Renamed the devicetree propertys ``voltage_reference`` and ``power_down_mode``
```
```suggestion
* Renamed the devicetree properties ``voltage_reference`` and ``power_down_mode``
```

rruuaanng replied:
In devicetree, usually, the term `property` is used instead of `properties`. And to be consistent with other and merged commits, I probably won't modify it, sorry Martin.

kartben replied:
The plural of property is indeed properties (not propertys) so yes, it should be changed :)

rruuaanng replied:
Oh! This involves my knowledge blind spot. I will modify it.
:------)

rruuaanng committed:
doc: Add a new migration guide entry to DAC

Rename the voltage_reference and power_down_mode
properties to voltage-reference and power-down-mode
in the devicetree and bindings of the DAC subsystem.

Signed-off-by: James Roy <rruuaanng@outlook.com>
